test/sqllogictest/types.slt

```diff
@@ -900,3 +918,13 @@ SELECT a::regtype FROM text_to_regtype ORDER BY a
 ----
 1082
 NULL
+
+query T
+SELECT 'date'::regtype::text::regtype
+----
+1082
+
+query T
+SELECT 'date'::regtype::text::regtype::text
```

Comment:
Also need to test the case where the regtype is an oid with no associated type, and the NULL case too. Same for regproc.

Reply:
I can add NULL. OID with no associated type is already handled on lines 867-890. There is no type associated with OID==1